sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RequestIdPolicy.java

```diff
@@ -15,13 +15,18 @@
  * the unique identifier for the request.
  */
 public class RequestIdPolicy implements HttpPipelinePolicy {
-    private static final String REQUEST_ID_HEADER = "x-ms-client-request-id";
+    private static final String REQUEST_ID_HEADER = "client-request-id";
+    private static final String LEGACY_REQUEST_ID_HEADER = "x-ms-client-request-id";
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
         String requestId = context.getHttpRequest().getHeaders().getValue(REQUEST_ID_HEADER);
```

Comment:
Not related to the changes being made in this PR, but a general question around the policy, could we add in another constructor which allows for a `Supplier<String>` to be passed which will be used in-place of `UUID.randomUUID()` if set.